### PR TITLE
Grep: Add grep dynamic executable

### PR DESCRIPTION
**Pull request by chococandy63**
Add the `grep` command of the grep command line utility.
Document usage of the `grep` command in `README.md`.

### CODE REVIEW

File: grep/README.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# Grep
+
+The `grep` command line tool is used for searching and matching text files contained in the regular expressions.
+
+
```
Comment (StefanJum) [inline grep/README.md:5]:
```suggestion

```
Don't use multiple consecutive blanc lines.

File: grep/README.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+Change directory into `run-app-elfloader` and run a similar command to the following:
+
+```console
+./run.sh -r path/to/dynamic-apps/grep /bin/grep [OPTION] [PATTERN] [FILENAME]
+```
+
+## Run the `grep` command on Linux
+
+To test the `grep` command run:
+
```
Comment (StefanJum) [inline grep/README.md:17]:
```suggestion
```

File: grep/README.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+```console
+
```
Comment (StefanJum) [inline grep/README.md:19]:
```suggestion
```

File: grep/README.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+/usr/bin/grep [OPTION] [PATTERN] [FILENAME]
+```
+
+For example, when in the current directory, run:
+
+```console
+/usr/bin/grep -i "hello world" demotext.txt
+```
+
```
Comment (StefanJum) [inline grep/README.md:28]:
```suggestion
```

File: grep/README.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+The `grep` command line tool is used for searching and matching text files contained in the regular expressions.
+
+
+## Running the grep Command Using [`run-app-elfloader`](https://github.com/unikraft/run-app-elfloader)
+
+Change directory into `run-app-elfloader` and run a similar command to the following:
+
+```console
+./run.sh -r path/to/dynamic-apps/grep /bin/grep [OPTION] [PATTERN] [FILENAME]
+```
```
Comment (StefanJum) [inline grep/README.md:12]: Give a snippet of what the output of the `run.sh` will be, both here and in the linux run section.

**Review by StefanJum — CHANGES_REQUESTED** [review]
Thanks a lot @chococandy63, looks great.
See the few comments, and please use your real name in the commit `signed-off-by` tag.

**Comment by chococandy63** [comment]
Hello @StefanJum 
I have made the necessary changes requested by you, kindly check!

File: grep/README.md [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+Change directory into `run-app-elfloader` and run a similar command to the following:
+```console
+./run.sh -r path/to/dynamic-apps/grep /bin/grep [OPTION] [PATTERN] [FILENAME]
+```
+For example, run:
+```console
+./run.sh -r ../dynamic-apps/grep /bin/grep -i "World" demotext.txt
+```
+Output:
```
Comment (StefanJum) [inline grep/README.md:15]:
```suggestion

Output:

```

File: grep/README.md [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+Booting from ROM..Powered by
+o.   .o       _ _               __ _
+Oo   Oo  ___ (_) | __ __  __ _ ' _) :_
+oO   oO ' _ `| | |/ /  _)' _` | |_|  _)
+oOo oOO| | | | |   (| | | (_) |  _) :_
+ OoOoO ._, ._:_:_,\_._,  .__,_:_, \___)
+             Prometheus 0.14.0~018fe1a4
+hello world!!!
+```
+## Run the `grep` command on Linux
```
Comment (StefanJum) [inline grep/README.md:27]:
```suggestion

## Run the `grep` command on Linux

```

File: grep/README.md [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+
+To test the `grep` command run:
+```console
```
Comment (StefanJum) [inline grep/README.md:30]:
```suggestion
To test the `grep` command run:

```console
```

File: grep/README.md [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+/usr/bin/grep [OPTION] [PATTERN] [FILENAME]
+```
+For example, when in the current directory, run:
```
Comment (StefanJum) [inline grep/README.md:33]:
```suggestion

For example, when in the current directory, run:

```

File: grep/README.md [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+```console
+/usr/bin/grep -i "hello world" demotext.txt
+```
+Output:
```
Comment (StefanJum) [inline grep/README.md:37]:
```suggestion

Output:

```

**Review by StefanJum — CHANGES_REQUESTED** [review]
Few more comments @chococandy63, hopefully the last ones.

**Comment by chococandy63** [comment]
@StefanJum  Thank you for the reviews, I have made the changes. Hopefully, everything is fine this time :)

**Comment by chococandy63** [comment]
@StefanJum  I pushed new changes(new commit) to the remote repository but I don't know how it got merged with this `grep` commit. Could you please guide me through this?

**Comment by StefanJum** [comment]
@chococandy63 first make a copy, make sure you have everything saved in case we mess something up.
Then:

```
git rebase -i HEAD~3 # <-- this will open a text editor with the commits
                                      # remove the line with the `file` commit (the lasts one probably)
                                      # save and exit

git log # <-- the `file` commit should be gone
git push --force
```

**Comment by chococandy63** [comment]
@StefanJum  Thank you! Looks fine to me, Can you please review this PR?

File: grep/README.md [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+
+For example, when in the current directory, run:
+
+```console
+/usr/bin/grep -i "hello world" demotext.txt
+```
+
+Output:
+
+```
```
Comment (StefanJum) [inline grep/README.md:49]:
```suggestion
```text
```

**Review by StefanJum — COMMENTED** [review]
One more comment @chococandy63, all good after that.

File: grep/README.md (hunk 11 above)
Comment (chococandy63) [reply grep/README.md]: @StefanJum  Done all the changes.

**Review by StefanJum — APPROVED** [review]
All good now, thanks a lot @chococandy63 
Reviewed-by: Stefan Jumarea <stefanjumarea02@gmail.com>
Approved-by: Stefan Jumarea <stefanjumarea02@gmail.com>

**Pull request merged**